### PR TITLE
Add description fields for each event type, License and Metadata.yaml

### PR DESCRIPTION
**Pull request by francoislg**

### CODE REVIEW

File: template.tpl [hunk 1]
```diff
@@ -2,7 +2,7 @@ ___INFO___
 
 {
   "displayName": "Coveo Analytics",
-  "description": "Logs an analytics event in Coveo Cloud",
```
Comment (dlafreniere) [inline template.tpl]: event*s*

File: template.tpl [hunk 2]
```diff
@@ -46,6 +46,60 @@ ___TEMPLATE_PARAMETERS___
...
+      },
+      {
+        "enablingConditions": [
+          {
+            "paramName": "eventType",
+            "type": "EQUALS",
+            "paramValue": "detailView"
+          }
+        ],
+        "displayName": "Detail View events are used in Coveo Analytics to define which product details are seen by the visitors.",
```
Comment (dlafreniere) [inline template.tpl]: I'm not sure it *has* to be product details. I could be document details or whatever

File: template.tpl [hunk 3]
```diff
@@ -131,7 +185,7 @@ ___TEMPLATE_PARAMETERS___
         "type": "TEXT"
       },
       {
-        "help": "",
+        "help": "The value used to differentiate your custom event for the given type",
```
Comment (dlafreniere) [inline template.tpl]: 🤔 this isn't super clear to me

File: template.tpl [hunk 4]
```diff
@@ -2,7 +2,7 @@ ___INFO___
 
 {
   "displayName": "Coveo Analytics",
-  "description": "Logs an analytics event in Coveo Cloud",
+  "description": "The Coveo Analytics template allows logging analytics event in Coveo Cloud",
```
Comment (dlafreniere) [inline template.tpl]: event *s*

File: template.tpl [hunk 5]
```diff
@@ -264,6 +330,21 @@ ___TEMPLATE_PARAMETERS___
...
+          {
+            "paramName": "eventType",
+            "type": "EQUALS",
+            "paramValue": "addToCart"
+          }
+        ],
+        "displayName": "Quantity",
+        "simpleValueType": true,
+        "name": "quantity",
+        "type": "TEXT"
```
Comment (dlafreniere) [inline template.tpl]: not number?

File: template.tpl [hunk 6]
```diff
@@ -264,6 +330,21 @@ ___TEMPLATE_PARAMETERS___
         "simpleValueType": true,
         "name": "brands",
         "type": "TEXT"
+      },
+      {
+        "help": "",
```
Comment (toussaints) [inline template.tpl]: Empty?

File: template.tpl (hunk 3 above)
Comment (francoislg) [reply template.tpl]: @fbeaudoincoveo any idea for this?

File: template.tpl (hunk 6 above)
Comment (francoislg) [reply template.tpl]: I just forgot to remove this one, it's from the example I did with you two this morning :|

File: template.tpl [hunk 7]
```diff
@@ -150,17 +204,29 @@ ___TEMPLATE_PARAMETERS___
...
+      },
+      {
+        "paramName": "eventType",
+        "type": "EQUALS",
+        "paramValue": "addToCart"
       }
     ],
-    "displayName": "Event metadata",
-    "name": "Detail View Event Metadata",
+    "displayName": "Event Metadata",
```
Comment (francoislg) [inline template.tpl]: Revert back to metadata

File: template.tpl [hunk 8]
```diff
@@ -47,6 +47,60 @@ ___TEMPLATE_PARAMETERS___
...
+        "enablingConditions": [
+          {
+            "paramName": "eventType",
+            "type": "EQUALS",
+            "paramValue": "custom"
+          }
+        ],
+        "displayName": "Custom events are configurable in any way you like to send data into Coveo Analytics.",
+        "name": "Custom Event Type Description",
+        "type": "LABEL"
```
Comment (fbeaudoincoveo) [inline template.tpl]: Oh cool I didn't know you could do that :)

File: template.tpl [hunk 9]
```diff
@@ -47,6 +47,60 @@ ___TEMPLATE_PARAMETERS___
...
+      },
+      {
+        "enablingConditions": [
+          {
+            "paramName": "eventType",
+            "type": "EQUALS",
+            "paramValue": "view"
+          }
+        ],
+        "displayName": "View events are used in Coveo Analytics to tell what pages your visitors are viewing.",
```
Comment (fbeaudoincoveo) [inline template.tpl]: Suggestion:

> Page view events are leveraged by Coveo ML Event Recommendation models (see <a href="https://docs.coveo.com/en/1886/">Coveo Machine Learning Event Recommendations Deployment Overview</a>).

File: template.tpl [hunk 10]
```diff
@@ -47,6 +47,60 @@ ___TEMPLATE_PARAMETERS___
...
+    "subParams": [
+      {
+        "enablingConditions": [
+          {
+            "paramName": "eventType",
+            "type": "EQUALS",
+            "paramValue": "custom"
+          }
+        ],
+        "displayName": "Custom events are configurable in any way you like to send data into Coveo Analytics.",
```
Comment (fbeaudoincoveo) [inline template.tpl]: Suggestion:

> Custom events can be leveraged in Coveo UA reports and by Coveo ML Event Recommendation models.

File: template.tpl [hunk 11]
```diff
@@ -478,7 +544,8 @@ ___TEMPLATE_PARAMETERS___
         "displayName": "Analytics endpoint",
         "simpleValueType": true,
         "name": "analyticsEndpoint",
-        "type": "SELECT"
+        "type": "SELECT",
+        "help": "The endpoint to use to contact your Coveo organization"
```
Comment (fbeaudoincoveo) [inline template.tpl]: I've updated this help text in my documentation review PR, so you may want to revert this change.

I don't have a strong opinion on this, though :)

File: template.tpl [hunk 12]
```diff
@@ -47,6 +47,60 @@ ___TEMPLATE_PARAMETERS___
...
+      },
+      {
+        "enablingConditions": [
+          {
+            "paramName": "eventType",
+            "type": "EQUALS",
+            "paramValue": "detailView"
+          }
+        ],
+        "displayName": "Detail View events are used in Coveo Analytics to define which detail pages or sections are seen by the visitors.",
```
Comment (fbeaudoincoveo) [inline template.tpl]: Suggestion:

> Product detail view events are leveraged by Coveo ML Commerce Recommendation models.

I'm not quite sure we should provide a label for this one yet, though.
